src/Autocomplete.js

```diff
@@ -0,0 +1,124 @@
+import React, { Component, PropTypes } from 'react';
+import cx from 'classnames';
+import constants from './constants';
+import Icon from './Icon';
+import Row from './Row';
+import Col from './Col';
+
+class Autocomplete extends Component {
+  constructor (props) {
+    super(props);
+
+    this.state = {
+      value: '',
+      minLength: this.props.minLength
+    };
+
+    delete this.props.minLength;
+
+    this.renderIcon = this.renderIcon.bind(this);
+    this.renderDropdown = this.renderDropdown.bind(this);
+    this.renderUl = this.renderUl.bind(this);
+    this._onChange = this._onChange.bind(this);
+  }
+
+  renderIcon (icon, iconClassName) {
+    return <Icon className={iconClassName}>{icon}</Icon>;
+  }
+
+  renderDropdown (data) {
+    const { value, minLength } = this.state;
+
+    if (minLength && minLength > value.length) {
```

Comment:
how about `if (minLength && minLength > value.length || !value) {` then below you could return the entire big one